app/controllers/missing.php

```diff
@@ -10,16 +10,87 @@ function index() {
         $this->load_view('missing/index');
     }
 
+    function validatePhone($phone) {
+        $regex = "/^((\+\d{0,3})?\d[\s-]?)?[\(\[\s-]{0,2}?\d{1,3}[\)\]\s-]{0,2}?\d{3}[\s-]?\d{4}$/i";
+
+        if( !preg_match( $regex, $phone)) {
+            return false;
+        }
+
+        return true;   
+    }
+
+    function validateDate($date) {
+        $dob = explode('-', $date);
+        if (count($dob) == 3) {
+            if (!checkdate($dob[1], $dob[2], $dob[0])) {
+                return false;
+            } 
+        } 
+        else {
+            return false;
+        }
+
+        return true;
+    }
+
+    function valid() {
+        if( !empty($_POST['person']['dob'])) {
```

Comment:
This is probably because the next line (line 39). We are not checking the return value of `validateDate`!